src/users/index.ts

```diff
@@ -18,35 +46,151 @@ export interface Users {
    */
-  assignRoles: (roleNames: string | string[], userId: string) => Promise<void>;
+  getAssignedRoles: (userId: string, opts?: GetAssingedRolesOptions) => Promise<Record<string, Role>>;
+
+  create: (userId: string) => Promise<string>;
+  delete: (userId: string) => Promise<boolean>;
+  rotateKey: (userId: string) => Promise<string>;
+  activate: (userId: string) => Promise<boolean>;
+  deactivate: (userId: string) => Promise<boolean>;
+  byName: (userId: string) => Promise<UserDB>;
+  listAll: () => Promise<UserDB[]>;
+}
+
+type GetAssingedRolesOptions = {
+  includePermissions?: boolean;
+};
+
+/** Operations supported for namespaced 'oidc' users.*/
+export interface OIDCUsers extends UsersBase {
   /**
-   * Revoke roles from a user.
+   * Retrieve the roles assigned to a user.
    *
-   * @param {string | string[]} roleNames The name or names of the roles to revoke.
-   * @param {string} userId The ID of the user to revoke the roles from.
-   * @returns {Promise<void>} A promise that resolves when the roles are revoked.
+   * @param {string} userId The ID of the user to retrieve the assigned roles for.
+   * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
-  revokeRoles: (roleNames: string | string[], userId: string) => Promise<void>;
+  getAssignedRoles: (userId: string, opts?: GetAssingedRolesOptions) => Promise<Record<string, Role>>;
 }
 
 const users = (connection: ConnectionREST): Users => {
+  const base = baseUsers(connection);
+
   return {
     getMyUser: () => connection.get<WeaviateUser>('/users/own-info').then(Map.user),
     getAssignedRoles: (userId: string) =>
       connection.get<WeaviateRole[]>(`/authz/users/${userId}/roles`).then(Map.roles),
+    assignRoles: (roleNames: string | string[], userId: string) => base.assignRoles(roleNames, userId),
+    revokeRoles: (roleNames: string | string[], userId: string) => base.revokeRoles(roleNames, userId),
+    db: db(connection),
+    oidc: oidc(connection),
+  };
+};
+
+const db = (connection: ConnectionREST): DBUsers => {
+  const ns = namespacedUsers(connection);
+
+  const allowCode = (code: number): ((reason: any) => boolean) => {
+    return (reason) => reason.code !== undefined && reason.code === code;
+  };
+
+  type APIKeyResponse = { apikey: string };
+  return {
+    getAssignedRoles: (userId: string, opts?: GetAssingedRolesOptions) =>
+      ns.getAssignedRoles('db', userId, opts),
     assignRoles: (roleNames: string | string[], userId: string) =>
+      ns.assignRoles(roleNames, userId, { userType: 'db' }),
+    revokeRoles: (roleNames: string | string[], userId: string) =>
+      ns.revokeRoles(roleNames, userId, { userType: 'db' }),
+
+    create: (userId: string) =>
+      connection.postNoBody<APIKeyResponse>(`/users/db/${userId}`).then((resp) => resp.apikey),
+    delete: (userId: string) =>
+      connection
+        .delete(`/users/db/${userId}`, null)
+        .then(() => true)
```

Comment:
Does work when the returned code is 204? That's what the server returns.

Reply:
`connection` will only throw errors for codes >= 400, so we'll be returning `true` for any 2** code here

https://github.com/weaviate/typescript-client/blob/b9bdb0c63922a91a2f5ab04f2348327be24dc69c/src/connection/http.ts#L373-L377